File: C++/Thread.cpp
#include <iostream>
#include <thread>
#include <vector>
#include <ctgmath>
#include "Header.hpp"

using namespace std;

// 9個のスレッドそれぞれで6回のループ
	// スレッドの内容を定義するこの関数をヘッダーファイルで定義してはならない
void mtAction(int m) {
	for (int n=1;n<7;n++) cout << "|(" << m << "," << n << ")| = " << std::hypot(m,n) << "\r\n";
}

void Thread(){

	cout << "同時に実行可能な数: " << std::thread::hardware_concurrency() << "\r\n";

	std::vector<std::thread> threads;

	for (int m=1;m<10;m++) threads.push_back(std::thread(mtAction,m));
	cout << "スレッドの終了を待ちます" << endl;
	for (std::thread &onethread : threads) onethread.join();
	cout << "スレッドは終了しました" << endl;

}

File: Rust/Legacy/Thread.rs
use std::thread;

pub fn main() {

	println!("\r\nこれからスレッドを試します\r\n");

	// 9個のスレッドそれぞれで6回のループ

	let mut threads:Vec<thread::JoinHandle<()>> = Vec::new(); // JoinHandleオブジェクトを格納する配列

	for m in 1..10 {
		let t=thread::spawn(move || {
			for n in 1..7 {
				println!("|({},{})| = {}",m,n,(m as f64).hypot(n as f64));
			}
		});
		threads.push(t);
	}

	println!("スレッドの終了を待ちます");
	for t in threads {
		t.join().unwrap(); // スレッドtの終了を待つ
	}
	println!("スレッドは終了しました");
	// 各々のスレッド内でのループの順番は守られるが,全54回のプロセスの順番はバラバラ
	// このコードを実行する毎に,順番は変化する

}